File: Online_Judge_Problems/boj/boj_11509_풍선_맞추기/boj_11509.js
/*-----------------------------------------------------
Sub  : [BOJ] 풍선 맞추기
Link : https://www.acmicpc.net/problem/11509
Level: 골드 5
Tag  : JS, greedy
Memo
-----------------------------------------------------*/

const TEST_MODE = true;

let input;

if (TEST_MODE) {
  const filePath = require('path').join(__dirname, 'input.txt');
  input = require('fs').readFileSync(filePath, 'utf-8').trim().split('\n');
}
else {
  input = require('fs').readFileSync(0, 'utf-8').trim().split('\n');
}


// input
const N = Number(input[0]);
const balloons = input[1].trim().split(' ').map(Number);
const arrowHeight = new Array(1000000).fill(0);
let answer = 0;

// processing
for (ball of balloons) {
  if (arrowHeight[ball] > 0) {
    arrowHeight[ball] -= 1;
    arrowHeight[ball - 1] += 1;
  } else {
    arrowHeight[ball - 1] += 1;
    answer += 1;
  }
}

// output
console.log(answer)

File: boj/boj_11509_풍선_맞추기/boj_11509.py
"""----------------------------------------------------
Sub  : [BOJ] 풍선 맞추기
Link : https://www.acmicpc.net/problem/11509
Level: 골드5
Tag  : Python, greedy
Memo
----------------------------------------------------"""

import sys

TEST_MODE = True


def get_inputs():
    if TEST_MODE:
        with open("input.txt", "r") as f:
            for line in f:
                yield line.rstrip("\n")
    else:
        for line in sys.stdin:
            yield line.rstrip("\n")


input_gen = get_inputs()


def input():
    return next(input_gen)


# input
N = int(input().strip())
balloons = list(map(int, input().strip().split(" ")))
arrow_heights = [0] * 1000001
answer = 0

# processing
for ball in balloons:
    if arrow_heights[ball] > 0:
        arrow_heights[ball] -= 1
        arrow_heights[ball - 1] += 1
    else:
        arrow_heights[ball - 1] += 1
        answer += 1

# output
print(answer)
